plugins/docker_binary_builder/app/models/binary_builder.rb

```diff
@@ -5,16 +5,26 @@ class BinaryBuilder
   ARTIFACTS_FILE_PATH = "/app/#{ARTIFACTS_FILE}".freeze
   DOCKER_HOST_CACHE_DIR = '/opt/samson_build_cache'.freeze
   CONTAINER_CACHE_DIR = '/build/cache'.freeze
+  TAR_LONGLINK = '././@LongLink'.freeze
+  PRE_BUILD_SCRIPT = 'pre_binary_build.sh'.freeze
 
-  def initialize(dir, project, reference, output)
+  def initialize(dir, project, reference, output, executor = nil)
     @dir = dir
     @project = project
     @git_reference = reference
     @output_stream = output
+    if executor
+      @executor = executor
+    else
+      @executor = TerminalExecutor.new(@output_stream, verbose: true)
+    end
   end
 
   def build
     return unless @project.try(:deploy_with_docker?) && File.exists?(File.join(@dir, DOCKER_BUILD_FILE))
+
+    run_pre_build_script if File.exists?(File.join(@dir, PRE_BUILD_SCRIPT))
```

Comment:
file location is used here and inside, would be drier / spread less knowledge when it is only handled inside

```
def run_pre_build_script
  file = File.join(@dir, PRE_BUILD_SCRIPT)
  return unless File.exist?(file)
  ...
end
```